app/src/renderer/model_factory.rs

```diff
@@ -37,24 +43,42 @@ impl ModelFactory {
      * Load all models.
      */
     pub fn load_models(&mut self) {
-        for model in MODEL_KEYS {
-            let raw_data = model.to_data();
+        for (model_name, texture_name) in TEXTURED_MODEL_DATA {
+            let raw_data = model_name.to_data();
             let string_data = from_utf8(raw_data).unwrap();
+
```

Comment:
'Load the model' and 'load the texture' are two snippets of code that could nicely be placed in helper functions.
With signatures like`AssetName -> Result<IndexedModel, &str>` (resp `AssetName -> Result<Texture, &str>`).

To make the code of 'load_models' more Rust-like as a whole, here is some inspiration: examples at https://doc.rust-lang.org/stable/rust-by-example/error/iter_result.html

Of course, this advice is nitpicky and not a blocker to merge this PR.

Reply:
Thanks for the docs link, allowed me to make this nice and pretty :smile: